### PR TITLE
Add breaking changes about jax.numpy.pad to change log

### PR DESCRIPTION
**Pull request by minoring**
https://github.com/google/jax/pull/5229#issuecomment-749028088

### CODE REVIEW

File: docs/CHANGELOG.rst [hunk 1]
```diff
@@ -24,6 +24,11 @@ jax 0.2.8 (Unreleased)
   * ``host_callback.outfeed_receiver`` has been removed (it is not necessary,
     and was deprecated a few months ago).
 
+* Breaking changes:
+
+  * ``jax.numpy.pad`` now take keyword arguments. Extra parameter (``constant_values``)
```
Comment (gnecula) [inline docs/CHANGELOG.rst:29]: I would propose "positional argument ``constant_values`` has been removed. ..."

File: jax/_src/numpy/lax_numpy.py [hunk 2]
```diff
@@ -2543,17 +2543,41 @@ def _pad(array, pad_width, mode, constant_values, stat_length, end_values, refle
     return _pad_empty(array, pad_width)
 
   else:
```
Comment (gnecula) [inline jax/_src/numpy/lax_numpy.py:2545]: I do not think that we need these changes, they were merged already in #5229.

**Review by gnecula — CHANGES_REQUESTED** [review]
In this particular case we were able to fix the breaking changes in the users' code so we did not revert your changes. If we did, the revert would have been visible in github also.

File: docs/CHANGELOG.rst (hunk 1 above)
Comment (minoring) [reply docs/CHANGELOG.rst]: Thanks for the proposal.
Updated! PTAL.

File: jax/_src/numpy/lax_numpy.py (hunk 2 above)
Comment (minoring) [reply jax/_src/numpy/lax_numpy.py]: Yes, I misunderstood.
I thought #5229 will be reverted and this new PR is merged after fixing the breaking changes.

I updated PR only modifying change log. PTAL:)

**Review by gnecula — APPROVED** [review]

**Pull request merged**